private/dir.go

```diff
@@ -0,0 +1,55 @@
+package private
```

Comment:
> I just don't like private as a oo like mean of "do not use this"

That’s fair; OTOH the semantics of what these utils do is very much targeted at `PolicyConfigurationIdentity` etc., very unlikely to be useful for anything else (well, `resolveExistingPathToFullyExplicit` could be useful, aka `canonicalize_file_name(3)`, but we don’t export _that_ one and it doesn’t belong into `containers/image` as a public API conceptually), and I have no intention of maintaining these functions as an API in any way useful to the general public for any other purpose than implementing `ImageTransport.PolicyConfiguration…`.  So the `private` subpackage name was intended to say “this is not for you, do not touch”; if language access controls can’t enforce it, we can still warn humans :)